### PR TITLE
add option to install dev kernel on clients

### PR DESCRIPTION
**Pull request by fullerdj**
TODO: reboot the clients with the new kernel after installation

### CODE REVIEW

**Comment by leseb** [comment]
please rebase on https://github.com/ceph/ceph-ansible/pull/1741 when merged.

**Comment by leseb** [comment]
@fullerdj what's the status of this?

**Comment by fullerdj** [comment]
@leseb The status is that it's now ready for review and merge if approved. Sorry for the delay.

File: roles/ceph-common/tasks/installs/install_debian_dev_kernel.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  package:
+    name: "{{ kernel_package_output.stdout.split()[0] }}"
+    state: latest
+    update_cache: yes
+  when:
+    - client_group_name in group_names
+    - ceph_dev_kernel
+
+- name: reboot to run ceph development kernel
+  shell: sleep 5 && shutdown -r now "Boot ceph dev kernel"
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_debian_dev_kernel.yml]: `sleep 5`?

File: roles/ceph-common/tasks/installs/install_redhat_dev_kernel.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - ceph_dev_kernel
+
+- name: configure grub to boot ceph kernel
+  command: grub2-mkconfig -o /boot/grub2/grub.cfg
+  when:
+    - client_group_name in group_names
+    - ceph_dev_kernel
+
+- name: reboot to install ceph development kernel
+  shell: sleep 5 && shutdown -r now "Boot ceph dev kernel"
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_redhat_dev_kernel.yml]: ditto

**Comment by leseb** [comment]
please rebase.

File: roles/ceph-common/tasks/installs/install_debian_dev_kernel.yml (hunk 1 above)
Comment (fullerdj) [reply roles/ceph-common/tasks/installs/install_debian_dev_kernel.yml]: Yes, this works around a known issue with ansible. See http://stackoverflow.com/a/39174307

File: roles/ceph-common/tasks/installs/install_redhat_dev_kernel.yml (hunk 2 above)
Comment (fullerdj) [reply roles/ceph-common/tasks/installs/install_redhat_dev_kernel.yml]: ditto

File: roles/ceph-common/tasks/installs/install_debian_dev_kernel.yml (hunk 1 above)
Comment (leseb) [reply roles/ceph-common/tasks/installs/install_debian_dev_kernel.yml]: Yeah I've actually figured that out when working on https://github.com/ceph/ceph-ansible/pull/1923 :)

**Comment by fullerdj** [comment]
rebased

**Comment by guits** [comment]
@fullerdj @batrick @leseb 
UP

**Pull request closed without merging**